lib/Emulation.cpp

```diff
@@ -157,27 +155,6 @@ const HistoryType& Emulation::history() const
   return _screen[0]->getScroll();
 }
 
-void Emulation::setCodec(const QTextCodec * qtc)
-{
-  if (qtc)
-      _codec = qtc;
-  else
-     setCodec(LocaleCodec);
-
-  delete _decoder;
-  _decoder = _codec->makeDecoder();
```

Comment:
Turns [out](https://github.com/lxqt/qterminal/issues/1143#issuecomment-2188740225) that `_decoder` here was keeping internal state(well, I'm kinda still guessing on this, since I didn't really dig into its internal(s) code) between the calls from below on line [250](https://github.com/lxqt/qtermwidget/pull/532/files#diff-0ce76997398f4a2c9df299136c2a5e7d44cfc4a410841b18ff2c5919b4032b84L250) and therefore in 1.4.0 it wasn't causing that qterminal [issue](https://github.com/lxqt/qterminal/issues/1143) whereby the replacement character aka `�` would appear in groups of 3 out of nowhere sometimes (typically on `btop`) which only happens in v. 2.0.0

Reply:
https://doc.qt.io/qt-6/qstringencoder.html → "The encoder remembers any state that is required between calls,..."